### PR TITLE
Upgrade to 3.2924.1575.g97389a9 / Add support for Visual Studio 2017

### PR DESCRIPTION
**Pull request by peters**

### CODE REVIEW

File: build.ps1 [hunk 1]
```diff
@@ -178,6 +182,10 @@ function Msvs
     $VXXCommonTools = $null
     $CmakeGenerator = $null
 
+	# https://github.com/Microsoft/vswhere/commit/a8c90e3218d6c4774f196d0400a8805038aa13b1 (Release mode / VS 2015 Update 3)
```
Comment (amaitland) [inline build.ps1]: My target is to strip all the binaries from this repository eventually (even pruning the old ones aka #35).

I'd prefer to download `vswhere` if required. Looks like there is a official release at https://github.com/Microsoft/vswhere/releases/tag/1.0.47 (same version was pushed to `nuget.org`).

File: build.ps1 [hunk 2]
```diff
@@ -197,6 +205,22 @@ function Msvs
             $VXXCommonTools = Join-Path $env:VS140COMNTOOLS '..\..\vc'
             $CmakeGenerator = 'Visual Studio 14'
         }
+		'v141' {
```
Comment (amaitland) [inline build.ps1]: Formatting - indentation is out (I know the whole `ps` file needs a reformat)

File: build.ps1 [hunk 3]
```diff
@@ -565,6 +596,7 @@ switch -Exact ($Target) {
         #VSX v110
 		VSX v120
         VSX v140
+		VSX v141
```
Comment (amaitland) [inline build.ps1]: How critical is it to include `VS2017` binaries in `cef.sdk`? `VS2017` has only just been released and as I'm sure your well aware they're only required when compiling `CefSharp` from source using `VS2017`. Official builds are still done using `VS2013`. Previously I'd stuck to just supporting two versions, `current` and `current - 1`. I'm a little reluctant to upgrade from `VC++2013` at this time. This also means a build machine would require three different `VS` versions. 

How big are the new packages compared to the previous ones?

File: build.ps1 [hunk 4]
```diff
@@ -14,7 +14,7 @@ param(
     [string] $CefBinaryDir = "../cefsource/chromium/src/cef/binary_distrib/",
 
     [Parameter(Position = 3)]
-    $CefVersion = "3.2883.1552.g88ff29a"
+    $CefVersion = "3.2924.1575.g97389a9"
```
Comment (amaitland) [inline build.ps1]: Upgrading of versions would ideally be in a separate `PR` as it's unrelated to adding `VS2017` support.

File: build.ps1 (hunk 3 above)
Comment (peters) [inline build.ps1]: VS2017 is ABI (binary) compatible with VS2015. This change is mainly for those that only have VS2017 installed.

File: build.ps1 (hunk 4 above)
Comment (peters) [inline build.ps1]: Hehe, sorry about! I was a bit lazy when submitting this PR. I can rebase and send a separate PR.

File: build.ps1 (hunk 3 above)
Comment (peters) [inline build.ps1]: I can comment this one if thats what you prefer. Probably not that many people have adopted VS2017 yet.

Comment (amaitland) [inline build.ps1]: > VS2017 is ABI (binary) compatible with VS2015. This change is mainly for those that only have VS2017 installed.

Do you mean `VC++`?

> I can comment this one if thats what you prefer. Probably not that many people have adopted VS2017 yet.

Yes please.

File: build.ps1 (hunk 4 above)
Comment (amaitland) [inline build.ps1]: No problem 👍 I'm hopeful a `2924` build will be available on http://opensource.spotify.com/cefbuilds/index.html in the next few days, so will upgrade to that `ASAP`.

 https://bitbucket.org/chromiumembedded/cef/commits/d3e47f59e9d3ba780cba5a77f90adf4f7ab5dc03

Comment (amaitland) [inline build.ps1]: Correction, I mean `2987`

**Comment by japj** [comment]
Hi, just wondering what the state of this PR is/what still needs be done to close this? (Can I help?)
I have been able to build cefsharp from VS2017 using the VS2015 toolset.

File: build.ps1 (hunk 1 above)
Comment (amaitland) [reply build.ps1]: Some code to check and download `vswhere` from a different script I wrote. Someone can adapt this if they wanted.

```
$programFilesDir = (${env:ProgramFiles(x86)}, ${env:ProgramFiles} -ne $null)[0]

$client = New-Object System.Net.WebClient;

$vswherePath = Join-Path $programFilesDir 'Microsoft Visual Studio\Installer\vswhere.exe'
#Check if we already have vswhere which is included in newer versions of VS2017 installer
if(-not (Test-Path $vswherePath))
{
    # Download vswhere if we don't have a copy
    $vswherePath = Join-Path $WorkingDir \vswhere.exe
    
    # TODO: Check hash and download if hash differs
    if(-not (Test-Path $vswherePath))
    {
	    $client.DownloadFile('https://github.com/Microsoft/vswhere/releases/download/2.2.7/vswhere.exe', $vswherePath);
    }
}
```

**Comment by perlun** [comment]
Agree, I would also find it interesting. @amaitland suggested this is the preferred start rather than #50, but this really needs a rebase (since the repository was cleaned up to get rid of old binaries.)

@peters Are you still around, would you have a chance to rebase this off of current master?

File: build.ps1 (hunk 3 above)
Comment (perlun) [reply build.ps1]: I'm quite certain that VC++ 2015 and 2017 are _not_ compatible with each other. That's never been the case before, there's _always_ been that requirement to upgrade with each new release.

I get the impression that this is still the case:

image
 
@peters I think we'll wait with VS2017 until CEF moves up to it. That's probably the easiest approach for now.

Comment (perlun) [reply build.ps1]: Ah, I just saw the comment in https://github.com/cefsharp/CefSharp/issues/1983 - seems like you're right @peters! My bad.

**Comment by perlun** [comment]
There are so many conflicts now that it's hard to really get an overview of this. @peters, some of what this PR aimed to do (upgrade CEF) has already been merged as of #53.

I will close this now. I think we will add VS2017 support using the approach suggested in #50. If there are still things that should be done, please do so and submit a new PR. Thanks for your effort nonetheless!

**Pull request closed without merging**